nixos/modules/services/x11/xserver.nix

```diff
@@ -313,12 +335,27 @@ in
 
       xrandrHeads = mkOption {
         default = [];
-        example = [ "HDMI-0" "DVI-0" ];
-        type = with types; listOf string;
+        example = [ "HDMI-0" { output = "DVI-0"; primary = true; monitorConfig = ""; } ];
+        type = with types; listOf (either
+          str
+          (submodule {
+            options.output = str;
+            options.primary = bool;
+            options.monitorConfig = lines;
```

Comment:
Well, the problem here is that it **should** fail for example if you provide a wrong option, unfortunately the only check that is done here is `isAttrs x || isFunction x`, so we need to fix that in the module system. Another problem this also has, is that you do the processing of it outside of the submodule, while the submodule has better ways to actually write the configuration using an internal option. This also would have the advantage that not every attribute needs to be specified.

For example your implementation fails if there is no `monitorConfig`.